Enhance shard legend tooltips

cachedout opened the pull request:
Refs: https://github.com/elastic/stack-monitoring/issues/35

## Summary
Adds additional information to the legend for shards in the Stack Monitoring application to make things more clear for a new user.

<img width="726" alt="screen shot 2019-02-22 at 09 54 48" src="https://user-images.githubusercontent.com/111616/53258374-bbc24000-3689-11e9-9d0a-76694a89bf16.png">


### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

~~- [] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~
- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
~~- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
~~- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios~~
~~- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)
- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2058/)

elasticmachine commented:
Pinging @elastic/stack-monitoring

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2140/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2160/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2419/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3269/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3282/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3502/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3511/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3525/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3535/)

cachedout commented:
All ready for review, @chrisronline

cachedout commented:
Polite review bump, @chrisronline :) Thanks.

chrisronline commented on x-pack/plugins/monitoring/public/components/elasticsearch/shard_allocation/lib/decorate_shards.js:
```diff
@@ -32,7 +32,11 @@ export function decorateShards(shards, nodes) {
       }
     }
-    return capitalize(shard.state.toLowerCase());
+    let retMsg = 'Shard ' + shard.shard + ' ' + shard.state.toLowerCase();
```
We probably need to localize this unfortunately. [This guide](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md#vanilla-js) might help but ping me if you have questions

cachedout replied:
Done! Thanks for the suggestion.

chrisronline reviewed (changes requested):
Going to officially request changes here. We need to add localization for the new text added.

cachedout commented:
@chrisronline I let this one lapse. I'll get back to it ASAP. Thanks for the bump!

elasticmachine commented:
Since this is a community submitted pull request, a Jenkins build has not been kicked off automatically. Can an Elastic organization member please verify the contents of this patch and then kick off a build manually?

cachedout committed:
Rebase to master and account for new directory layout

cachedout committed:
Rebase to master and account for new directory layout

cachedout committed:
i18n

cachedout committed:
i18n

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/5780/)

streamich committed:
Plugin mocks (#40123)

* test: 💍 add ability to mock plugins in ui/new_platform

* chore: 🤖 delete olds NP backdoor mock

* test: 💍 normalize plugin contract naming in line with Core

* test: 💍 use mock from ui/new_platform instead of core

* test: 💍 remove ui/new_platform mocking from Core

* chore: 🤖 leave only one mocking strategy for ui/new_platform

* style: 💄 format according to Prettier

* chore: 🤖 remove /src/plugins/__mocks__ folder

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/5783/)

cachedout commented:
@chrisronline I believe this is ready for a re-review. Thanks!

mshustov committed:
Expose elasticsearch error wrapper (#40242)

* expose elasticsearch error wrapper

* generate docs

* Update src/core/server/elasticsearch/errors.ts

Co-Authored-By: Rudolf Meijering <skaapgif@gmail.com>

* address Oleg comments

walterra committed:
[ML] Data Frames: Fix transform list refreshes. (#40348)

The custom hook introduced in #40129 had issues with triggering multiple times in nested components and would cause multiple request of the same data. This PR improves the hook to avoid this behaviour:

- Similar to how you could pass a prop like onChange() to a React component, you can now pass a callback for isLoading() and onRefresh() to the custom hook.
- This allows more fine grained event handling. Observable subscribers will only be set up if they are actually needed. Previously we were creating them whenever the hook was initialized. The hook takes care of subscribing/unsubscribing.
- As you can see in the diff, the code for using the hook in a component became much simpler. It's no longer necessary to use multiple different calls using useEffect().
- The expanded rows of the transform list had two issues: High expanded rows could be cut off and flickering could occur in the message tab and preview tab tables. This PR fixes it by using SCSS overrides as a last resort to disable the expanded row animations. These fixes allow us to increase the maximum audit messages to 50. The order of audit messages in this PR is now ascending to reflect the order used in the anomaly jobs table.

ogupte committed:
[APM] Update default index and identifiers to match feature name (APM Agent Configuration) (#40224)

* [APM] removes apm agent config index from list of indices that match the
apm index patter. Fixes smoke tests.

* [APM] rename the index .apm-cm -> .apm-agent-configuration and many
code identifiers to match the feature name (APM Agent Configuration)

* updated remaining references from 'config management' to 'agent configuration'

vitalics committed:
[TSVB] [Unit] Increase Gauge coverage (#38966)

*  Increase Gauge coverage

dmlemeshko committed:
Run more tests in firefox (#40140)

* [smoke suite] add code search tests

* run FF xpack tests 5x times

* [smoke suite] add more x-pack tests

* Revert "run FF xpack tests 5x times"

This reverts commit c59046f43d5f61d0b5e083d185c22eff850fba49.

dmlemeshko committed:
[ftr] replace getProperty with getAttribute (#40358)

* [ftr] replace getProperty with getAttribute

* fix failing tests

* [services/pipeline_editor] revert change

Commit:
[Code] Add ctags icon (#40263)

thirugnanamk89 committed:
curl related doc changes in load dataset tutorial (#16040)

* curl related doc changes in load dataset tutorial

* Correcting typo and incorporating review comments

* Fixing new line issues and corrected example commands

* Removing '#' as it seem to combine multiple lines

* Update tutorial-load-dataset.asciidoc

* Removing the template command

* Remove `doc` path

walterra committed:
[ML] Fix cell formatting. (#40396)

- Fixes a regression to restore date formatting for the pivot preview table.
- EUI's dataType attribute for table columns is now used where applicable to provide better formatting for certain field types.
- The rendering of cell values has been align for both the source and the pivot preview table.

mshustov committed:
Fake request interface compatible with Kibana Request (#40363)

* make FakeRequest compatible with http request

* FakeRequest compatible with Http Request

alexwizp committed:
[TSVB] Apply dateHistogramInterval utility (#39665)

* [TSVB] Apply dateHistogramInterval utility

* isModelValid -> isModelInvalid

mshustov committed:
Env contains distributable flag (#40361)

* env contains distributable flag

* make sure flag is always boolean

azasypkin committed:
Include new platform x-pack plugin paths to Jest roots. (#40422)

Commit:
Fix issue with long values in TSVB static metric (#40256)

* Fix issue with long values in TSVB static metric

* Handle numeric values with decimal zeros

* Work properly with exponential values

* Wrap into boolean

thomasneirynck committed:
[Maps] Use default locale as fallback for EMS tile layers (#40433)

Commit:
[kbn/es] add command to build es snapshots for each license an… (#40331)

XavierM committed:
Remove warning euiBasicTable on host tab (#40447)

* remove warning euiBasicTable on host tab

* fix types

Commit:
Added ILM smoke test to ensure the app has loaded and a data test subject for the section header. (#39859)

Commit:
Added snapshot restore smoke test to ensure the app has loaded. (#39812)

* Added snapshot restore smoke test to ensure the app has loaded.

* Fixed linting issues.

* Updated test to reflect changes to Snapshot and Restore's name.

* Fixed merge update.

monfera committed:
Fix: restore special case check dropped in refactoring (#40369)

Bamieh committed:
[Core] Remove lock file from packages/kbn-spec-to-console and… (#40395)

* remove lock file from packages/kbn-spec-to-console and update jest

* update handlebars to 4.1.0

* update jest-cli

Commit:
[Code]  cleanup repo object  (#40356)

[Code] workspace repo clean up

Commit:
Added index management app loading test. (#40219)

Commit:
[Code] Register definition handler for languages supported by ctags (#40472)

* [Code] Register definition handler for languages supported by ctags

* Refactor catgs support langugae

maryia-lapata committed:
[Vis Editor] Agg params compressed (#40137)

* Apply compressed for EUIFormRow controls

Commit:
Bugfix: Improve filter sorting logic to be more stable (#40246)

* Improve filter sorting logic to be more stable

* Improve filter sorting logic to be more stable

* test

* Code review improvement (use shorthand non-null assertion)

* code review

walterra committed:
[ML] Data Frames: Fix form validation (#40427)

Fixes the form validation of transform id and destination index name.

- In addition to checking against existing transform ids, the input is now validated according to the data frame API docs
- The destination index input field is now validated against the index name limitations found in Elasticsearch's documentation. Since these rules are too much to be added as a helper text, a link to the docs is part of the helper text.
- If the destination index name matches an existing index, the Next button is no longer disabled, but the helper text for the input field states that creating this transform will modify the existing destination index.

dgieselaar committed:
[APM] Persist table parameters to URL (#39912)

If the user changes the table options (page index, size, sort), they are reflected in the URL. This means that the user can navigate back and forth without losing the table state.

Closes #39340.

PhilippBaranovskiy committed:
Replaced not-working search settings live update with EuiDelayRender component (#39987)

dgieselaar committed:
[APM] Migrate deprecated date_histogram parameter (#39886)

* [APM] Migrate deprecated date_histogram parameter

Closes #39584.

Replaces 'interval' with 'fixed_interval'

* Update snapshot

mikecote committed:
Use duration syntax for alert intervals (#40193)

* Use duration syntax for alert intervals

* Add tests to ensure route validates regex

* Add missing test

XavierM committed:
fix style of the draggable in siem app (#40454)

angorayc committed:
fix style for persistent kql bar (#40487)

nreese committed:
[Maps] fix react typing for styles (#40518)

stephmilovic committed:
Siem query review (#40279)

nreese committed:
[Maps] use custom labels in legend details (#40308)

* [Maps] use custom labels in legend details

* use dynamic label in vector style field select

* review feedback

Commit:
fix(code/frontend): fix top bar layout (#40342)

FrankHassanabad committed:
[SIEM] Anomaly UI table changes (#40440)

## Summary

- [x] Changed "Score" to "Anomaly Score"
- [x] Changed "Detector" to be "Job Name"
- [x] Added Link from "Job Name" to Anomaly Explorer page
- [x] Aligned text of the tables to be at the top
- [x] Removed the Information I from the table rows but kept it on the Host Details and Network Details
- [x] Added Timestamp to the end of the table
- [x] Moved "Job Name" to be after "Anomaly Score"
- [x] Removed Host Name from Anomalies table when on the Host Details page as it is redundant
- [x] Removed Network Name from Anomalies table when on the Network Details page as it is redundant
- [x] Added anomaly score Default threshold of 50 for the advanced settings page

Advanced setting for default Anomaly Score:
<img width="1225" alt="Screen Shot 2019-07-05 at 6 15 31 PM" src="https://user-images.githubusercontent.com/1151048/60749093-5abd4980-9f52-11e9-9340-08ef8e462c8f.png">

Before Host Overview:
<img width="2192" alt="before-overview-hosts" src="https://user-images.githubusercontent.com/1151048/60746932-23916d00-9f3f-11e9-81fb-e3dba98af160.png">

After Host Overview:
<img width="2186" alt="after-overview-hosts" src="https://user-images.githubusercontent.com/1151048/60746938-2f7d2f00-9f3f-11e9-9a4c-37f5bbc19771.png">

Before Host Details:
<img width="2201" alt="before-host-details" src="https://user-images.githubusercontent.com/1151048/60746961-4f145780-9f3f-11e9-9086-2709b7957221.png">

After Host Details:
<img width="2202" alt="after-host-details" src="https://user-images.githubusercontent.com/1151048/60746969-56d3fc00-9f3f-11e9-9110-5fb46fb398c9.png">


Before Network Overview:
<img width="2199" alt="before-network-overivew" src="https://user-images.githubusercontent.com/1151048/60746954-41f76880-9f3f-11e9-8c75-cc7e6dbde276.png">

After Network Overview:
<img width="2196" alt="after-network-overview" src="https://user-images.githubusercontent.com/1151048/60746957-47ed4980-9f3f-11e9-843a-a2b210347649.png">

Before Network Details:
<img width="2200" alt="before-network-details" src="https://user-images.githubusercontent.com/1151048/60746972-5d627380-9f3f-11e9-8dcb-cc1e1d73c0f9.png">

After Network Details:
<img width="2189" alt="after-network-details" src="https://user-images.githubusercontent.com/1151048/60746974-63585480-9f3f-11e9-9847-4645a7b1ab1d.png">

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [x] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials
- [x] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)

### For maintainers

- [x] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)
~- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~

Commit:
[Code] enable multinode test (#40563)

Commit:
[Code] don't crash frontend if language server returns a empty value. (#40570)

Commit:
[Code] fix index hang when proxy is marked unusable (#40485)

andrewpatroni committed:
Update documentation_links.ts (#40333)

https://www.elastic.co/guide/en/kibana/7.2/code-install-language-server.html provides 404, the correct link is https://www.elastic.co/guide/en/kibana/7.2/code-install-lang-server.html

Commit:
[Code] update ctags lsp version and remove unsupported langs (#40572)

peteharverson committed:
[ML] Fixes results time range if anomalies occur after data stops (#40503)

* [ML] Fixes results time range if anomalies occur after data stops

* [ML] Edits following review

dgieselaar committed:
[APM]: use correct interval field key for histogram aggregation (#40598)

Fixes an issue with distribution aggs failing because #39886 inadvertently changed the `interval` key for a histogram agg to `fixed_interval`, which only works on date_histogram aggs.

alvarezmelissa87 committed:
[ML] Data frame - ensure pivot preview works with comma-separated index patterns  (#40282)

* split comma-sep indexPatterns into array for request

* always check for comma-separated index pattern

* fix type error

* types cleanup

justinkambic committed:
[Uptime] Fix unsigned integer parsing error and add tests (#40526)

* Fix unsigned integer parsing error and add tests.

* Fix new defect, improve serialization function and add another test.

* Fix redundant code.

* Remove redundant function call.

peteharverson committed:
[ML] Fixes anomalies table sort behavior when sorting on severity (#40600)

walterra committed:
[ML] Data Frames: Fixes start/stop/delete for failed transforms. (#40532)

- Fixes handling of failed transforms according to #40298 (comment). This adapts the API endpoints to allow force/wait_for_completion flags where applicable.
- Renamed DATA_FRAME_RUNNING_STATE to DATA_FRAME_TASK_STATE to better reflect the API naming.
- #40129 introduced an observable to refresh the transform list. The transform list actions now make use of this too and no longer require getJobs() to get passed around as a deeply nested prop.
- The state handling for overall loading and errors for the transform list is fixed/improved by this PR.
- On initial load, the list no longer shows No data frame transforms found, only the loading indicator.
- If loading the transform list data fails, the list gets replaced by an error-callout to avoid displaying out-of-date data and access to actions which might not work anymore.
- Fixes a bug where the transform list would no longer pick up refresh triggers after the request returned an error.
- Failed state in in the transform list uses now the danger color for the badge and adds a tooltip with the text provided in the reason field of the transform's stats.
- Fixes a regression where the messages pane would no longer load.

nreese committed:
[maps] fix symbol anchor position (#40524)

hbharding committed:
[APM] fixed header offset to prevent scrollbar from showing (#40147)

* fixed header offset to prevent scrollbar from showing

* remove topNavHeight variable (not needed)

walterra committed:
[ML] Data Frames: Fixes search triggering onChange on every keystroke. (#40607)

- Replaces EuiFieldSearch with EuiSearchBar for the data frames transform wizard. EuiSearchBar has an incremental prop, when set to false, onChange will only trigger on enter instead on every keystroke.
- Fixes a bug where no search results would trigger an error callout for the source data table.

Commit:
Removed docs of deleted directive (#40597)

dgieselaar committed:
[APM]: Fix transaction breakdown layout on IE11 (#40615)

Closes #40354.

zfy0701 committed:
[Code]: fix  mocha, jest test file path (#40574)

mw-ding committed:
[Code] always read file content from the original bare repo in indexer (#40176)

* [Code] reenable code mocha tests

* skip multi node test

* [Code] read file content from the original bare repo if fails to open workspace in indexr

* More refactoring

* Add more unit tests

* Add unit tests

* always read file content from the original bare repo

* remove catching the workspace prepartion errors in indexer

XavierM committed:
[SIEM] Enhance inspect modal  (#40530)

* Enhance inspect modal to have a statistic tab and only show the body of the request to allow user to use it in dev tools

* review I

* review II

simianhacker committed:
[Infra UI] Fixes #39809 - Sync state with localStorage for Metrics Explorer (#40099)

* Sync state with localStorage for Metrics Explorer

* Fixing typing issues for useStateWithLocalStorage

* Adding localStorage support

* Adding tests

* ESLint Fixes

mshustov committed:
Expose "is TLS enabled" flag for Kibana HTTP Server (#40336)

* expose it TLS enabled on Http server

* setup contract is under one section in tests

* regenerate docs

* isTLSEnabled --> isTlsEnabled

mikecote committed:
Prevent alert intervals from being 0 (#40634)

* Prevent intervals from being 0

* Add integration test

KOTungseth committed:
[DOCS] Create Canvas workpads (#38616)

* Create workpads (in Canvas)

* Continuing work on intro

* Share your workpad content

* Reorganization

* Added GIFs

* Final clean up

* Comments from reviewers

* Comments from Gail

nreese committed:
[Maps] create visual regression test for vector styling point with icons (#39363)

* [Maps] create functional test for vector styling point with icons

* clena up

* use full screen to hide kibana chrome

* collapse legend to limit UI surface area

thomasneirynck committed:
[Maps] Remove redundant .setData call (#40628)

mshustov committed:
createCluster requires a partial elasticsearch config (#40405)

* createCluster created with partial config

* add tests

* re-genereate docs

igoristic committed:
Issue 36738: Added zoomout functionality to all charts (#39072)

* Added zoomout functionality to all charts

* Assigning timeout to its variable

* Simplified the logic a little

* Will move this fix to a different PR

* Addressed feedback

* Addressed feedback

* Added more details about the 10ms value

* Added i18n and unit tests

gchaps committed:
[DOCS] Updates Watcher docs to match updated UI (#40106)

* [DOCS] Updates Watcher docs to match updated UI

* [DOCS] Incoporates watcher review comments and adds badge

* [DOCS] Updated watcher security doc based on feedback

* Update docs/management/watcher-ui/index.asciidoc

Co-Authored-By: Lisa Cawley <lcawley@elastic.co>

stephmilovic committed:
[SIEM] Fix url state (#40653)

[SIEM] Fix url state (#40653)

tsullivan committed:
[Reporting/Telemetry] Rewrite telemetry collection to use a single query (#34327)

* prettier formatting

* new get reporting usage module

* some typescript

* types using mapped types and intersection types

* fix tests

* fix more tests

* more tests

* self-review

* add filterPath to query

* self review 2

* check feature availability in getDefaultObject

* add data modeling unit test

* fix lint

* rename a couple files

* remove defaultsDeep usage

* polish decorate range stats

* remove hardcoding for known export types

* fix unit test

* Rename a type to ExportType

* remove param used for analyzing raw hits

* remove obsolete ts hint

* isReady fix

* fix misplaced files

justinkambic committed:
[Uptime] [Test] API Test update (#40217)

* Extract monitor page title query string to exported string const.

* Add test for monitor page title.

* Extract MonitorStatesQuery string from gql wrapper.

* Write tests for snapshot histogram graphql endpoint.

* Write tests for monitor states graphql endpoint.

* Import new tests to graphql describe block.

* Update bad fixture for monitor states query.

* Update bad fixture for monitor states filtered by id.

pmuellr committed:
Validate nodemailer action service values (#40675)

This adds validation when using the `service` config value for the `.email`
action type, against the list of services nodemailer supports.

Also adds support for the nodemailer `jsonTransport` option, using the
synthetic service `__json`, for test/demo usage.

Zacqary committed:
[Logs UI] Fix inaccuracy when jumping to a faraway time target (#40303)

stephmilovic committed:
clear unique key prop err for tab nav (#40686)

tsullivan committed:
Fix Relative Date format in CSV export (#40544)

* Fix Relative Date format in CSV export

* fix date nanos

* functional test

mattapperson committed:
refactor startTestServers for individual start and stop (#40545)

* refactor startTestServers for individual start and stop

Commit:
Fix Graph license checks (#40580)

maryia-lapata committed:
Import ui/angular-bootstrap (#40508)

dgieselaar committed:
[APM] Make sure stacked area charts handle no data points inco… (#40353)

* [APM] Make sure stacked area charts handle no data points incorrectly

Closes #40351.

* Refactor CustomPlot/TransactionCharts

- [x] Keep API calls in a separate file from the useTransactionBreakdown hook
- [x] Bail out of render function early when `receivedDataDuringLifetime` is false
- [x] Use `cluster` prop to draw lines over area instead of creating a new plot
- [x] can lineseries be stacked without creating a new plot?
- [x] Move functionality of SyncChartGroup to ChartsTimeContext (renamed to ChartsSyncContext)
- [x] Only determine noHits inside of CustomPlot instead of using hits.total
- [x] Remove use of getEmptySeries except for testing
- [x] Send TimeSeries from the API instead of mapping data inside of the component
- [x] Use default data in useTransactionBreakdown to prevent is-null checks
- [x] Remove usage of start, end in CustomPlot

* Additional cleanup on aisle five

* Don't show GridLines or YAxis component if there is no data

* Make formatters less defensive

* Fix loading state for TransactionCharts

* Update translation files

Commit:
chore(code/frontend): refactor `Route` component (#40500)

sorenlouv committed:
[APM] Distinguish between loading state and empty state (#40651)

* [APM] Distinguish between loading state and empty state

* Remove async describe

* Fix tests

* Show empty state outside table for agent configuration list

* Fix translations

alvarezmelissa87 committed:
ensure fieldCaps response is not undefined (#40682)

markov00 committed:
Configure GraphiteURLs whitelist from kibana.kml (#39480)

dgieselaar committed:
[APM] Agent config: fix typo in validation error message for e… (#40727)

Closes #40726.

kertal committed:
De-angularize kbn_doc_view JSON view  (#39515)

Replace Angular code by React + EuiCodeEditor

legrego committed:
Role Management - use ES Builtin Privilege API to drive list o… (#40270)

* use ES builtin privileges API for role management

* Exclude 'none' from privilege lists

* additional cleanup

clintandrewhall committed:
Move and consolidate types (#40701)

nreese committed:
[Maps] redux file structure cleanup (#40655)

* [Maps] redux file structure cleanup

* update path for connected_components folder in scss include

Commit:
[Mappings editor] Initial JSON version (#40415)

Zacqary committed:
[Inffra UI] Widen Group By dropdown and add tooltips to custom metrics (#40692)

Zacqary committed:
[Logs UI] Add a time cursor to the log minimap (#39538)

* Add a time cursor to the log minimap

* Fix type checking

sorenlouv committed:
[APM Tutorial] Fix APM Server link on Cloud (#40711)

* [APM Tutorial] Fix APM Server link on Cloud

* fix translations

Commit:
Fix usage of Filters aggregation by Terms' Other bucket (#40698)

simianhacker committed:
Fixes #40138 - Convert bytes to bits before formatting for bits (#40523)

* Fixes #40138 - Convert bytes to bits before formatting for bits

* Adding comment

* Fixing test

* Changing createDataFormatter to createBytesFormatter

* Removing tests for old bytes formats

* fixed logic typo

mw-ding committed:
[Code] fallback with symbolInformation.name for typeahead (#40654)

markov00 committed:
Upgrade @elastic/charts dependency to 7.2.1 (#40751)

Commit:
[DOCS] Fixes typo in discover tutorial (#40785)

chrisronline reviewed (approved):
LGTM! Nice work!

simianhacker committed:
Fixes #38130 - Fix CSS to improve IE11 experience (#40309)

* Fixes #38130 - Fix CSS to improve IE11 experiance

* Increasing the padding to match top bar margins

Commit:
Added app load test for Cross Cluster Replication. (#40708)

Commit:
[ts] upgrade to 3.5.3 (#40228)

* [ts] upgrade to 3.5.2

* [ts] run from cwd so that relative paths are correct

* move eslint-disable-line comment into jsx attribute

* autofix eslint violations

* avoid generic type, it's not necessary and problematic

* make elasticsearch.cluster optional, many instances don't have them

* remove invalid prop

* expand AllowUnknownProperties to cleanly handle arrays

* esfilter values can sometimes be an array of strings

* allow exception objects to have unknown properties

* define accumulator as a boolean

* fix return type

* return a 404 if beat isn't found after update

* use Object.values for better types

* define return type of get() call

* define value type for Set

* define return value of get()

* define State property type

* use less get(), so ts can infer types

* define Set item type

* map state type

* make default_operator optional, since it's not always defined

* remove seemingly unused prop

* define return type of get() fn

* define inner type for state

* don't define base types are objects with index signatues

* regenerate public api docs

* make indices privileges optional too

* remove unnecessary index-signature from Exception interface

* use variadic _arg instead

* [core/plugin] use Record<string, any> rather than {}

* replace a couple more instances of {}

* revert some unnecessary changes

* remove unused types

* [reporting] output, payload, and meta are required properties

* bump to latest patch version

peteharverson committed:
[ML] Fixes time ranges shown for jobs in job selector flyout (#40757)

mikecote committed:
Ability to enable and disable alerts (#40543)

* Ability to enable / disable alerts

* Add docs

* Fix broken jest test

* ESLint fixes

* Remove enabled parameter from update API, create 2 new endpoints

* Fix ESLint errors

lukasolson committed:
Cancel Discover fetch requests if a new one is submitted (#39963)

* Cancel discover fetch requests if a nnew one is submitted

* Add comment

* Remove unused translations

FrankHassanabad committed:
[SIEM] Adds conditional linking within the application for machine learning jobs (#40547)

## Summary

Fixes these issues when drilling down from an ML job to a SIEM page:

* An influencer or entity from a variable such as $host.name$ could be a comma separated value
* An influencer or entity from a variable such as $host.name$ could be still a variable and not replaced in which case it causes invalid URL's and KQL
* You could end up with multiple influencers or entities when you drill down to a link such as host details we have more than one host name where it would be better to redirect to the host overview and show the multiple hosts as part of the KQL.
* https://github.com/elastic/ingest-dev/issues/564

You prefix your jobs with the label of `ml-hosts` or `ml-network` instead of `hosts` or `network` and allow the router to re-direct depending on what is given from the machine learning.

<img width="1230" alt="Screen Shot 2019-07-10 at 1 06 48 AM" src="https://user-images.githubusercontent.com/1151048/60947979-24affa80-a2af-11e9-8c9c-3790beb08db4.png">
 
If you have multiple hosts then it will re-direct you to the hosts overview page instead of the details page with all of the hosts separated by OR clauses. Likewise if you have multiple networks it will redirect you to the network page with `source.ip` `destination.ip` or'ed together with each network IP. Both of those will keep the AND clause of any influencers or entities you carry with it:

<img width="775" alt="Screen Shot 2019-07-10 at 12 55 36 AM" src="https://user-images.githubusercontent.com/1151048/60948138-76f11b80-a2af-11e9-9f74-e5566cb74d4b.png">

<img width="1214" alt="Screen Shot 2019-07-10 at 12 55 50 AM" src="https://user-images.githubusercontent.com/1151048/60948120-68a2ff80-a2af-11e9-8813-6138e34df550.png">

If you have any variables that are not flushed with values it will take you to the overview pages as well and not try to keep you on a details page with an dollar signed value. 

## Caveats

* This does not have KQL support for operators such as these `<`, `<=` or `>` mixed in. It only checks and works with the `and`, `or`, and `not`
* This only works with a very specific subset of KQL in which it expects the values to be wrapped in double quotes such as `host.name: "host-value"`
* This is tested with the soon to be shipped SIEM jobs and not with ad-hoc created jobs.
* The approach is that it tries to _only_ change or manipulate the URL's RISON and KQL if it has to manipulate them. If it does not detect any comma separated values or any variables not replaced it will leave the RISON and KQL alone. 

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
~- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~
- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials
- [x] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
~- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~

### For maintainers

~- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~
- [x] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

bmorelli25 committed:
[docs][APM] Agent config docs (#40084)

jen-huang committed:
Fix Rollup tag not showing on rollup index patterns in list (#40297)

lcawl committed:
[DOCS] Fixes typo (#40668)

Commit:
[Code] parse and register language file extension from monaco-editor (#40578)

Commit:
fix(code/frontend): HEAD should be default branch in branch selector (#40832)

mshustov committed:
expose ES createClient to plugins (#40717)

cachedout committed:
Merge branch 'shard_tooltips' of github.com:cachedout/kibana into shard_tooltips

cachedout commented:
Guh. Rebase gone bad.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/6817/)
